pom.xml

```diff
@@ -200,10 +200,11 @@
                 <version>${jayway-version}</version>
             </dependency>
             <dependency>
-                <groupId>commons-lang</groupId>
-                <artifactId>commons-lang</artifactId>
-                <version>${commons-lang.version}</version>
+                <groupId>net.minidev</groupId>
+                <artifactId>json-smart</artifactId>
```

Comment:
As per this PR on jsonPath: https://github.com/Netflix/dgs-framework/pull/60
it is possible to use jayway with only jackson without json-smart.

https://github.com/Netflix/dgs-framework/issues/58
this issue also indicates that json-smart is not well maintained. Do you think it would be possible to use jackson instead? Would that be feasible and would it require a lot of cjange?

Reply:
Other relevent links that I found: https://github.com/json-path/JsonPath/blob/master/README.md#jsonprovider-spi

https://github.com/json-path/JsonPath/issues/676

Reply:
I think switching from json-smart (default json provider for json-path) to jackson would require some changes and would be worth an issue on its own.

@authorjapps : Do you support switching to jackson?

Information about supported json providers for json-path: https://github.com/json-path/JsonPath

json-smart itself isn't really badly maintained, the last version dates from July 2023.

Reply:
I switched to Jackson as JSON provider for json-path in a single unit test (ZeroCodeAssertionsProcessorImplTest). After that nearly half the tests fail (in the screenshot I filtered out the passed tests). It seems json-path is not agnostic to the JSON provider it uses.

[jsonpath_with_jackson_patch.txt](https://github.com/authorjapps/zerocode/files/14628415/jsonpath_with_jackson_patch.txt)

![failed_tests_jsonpath_with_jackson](https://github.com/authorjapps/zerocode/assets/65255151/38d9b9da-ad2e-4889-acbf-9d0433b6e8ca)

Reply:
@baulea @authorjapps  I think this is not a jsonpath issue. There is some code we have that breaks JsonPath abstraction to work on json arrays. That is the reason all the array tests are failing. 

As @baulea mentioned, we can look into this as a separate issue and merge this for now

Reply:
created issue for the same #648